apis/tempo/v1alpha1/tempomonolithic_types.go

```diff
@@ -0,0 +1,265 @@
+package v1alpha1
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TempoMonolithicSpec defines the desired state of TempoMonolithic.
+type TempoMonolithicSpec struct {
+	// Storage defines the backend storage configuration
+	//
+	// +kubebuilder:validation:Optional
+	Storage *MonolithicStorageSpec `json:"storage,omitempty"`
+
+	// Ingestion defines the trace ingestion configuration
+	//
+	// +kubebuilder:validation:Optional
+	Ingestion *MonolithicIngestionSpec `json:"ingestion,omitempty"`
+
+	// JaegerUI defines the Jaeger UI configuration
+	//
+	// +kubebuilder:validation:Optional
+	JaegerUI *MonolithicJaegerUISpec `json:"jaegerui,omitempty"`
+
+	// ManagementState defines whether this instance is managed by the operator or self-managed
+	//
+	// +kubebuilder:validation:Optional
+	Management ManagementStateType `json:"management,omitempty"`
+
+	// Observability defines observability configuration for the Tempo deployment
+	//
+	// +kubebuilder:validation:Optional
+	Observability *MonolithicObservabilitySpec `json:"observability,omitempty"`
+
+	// ExtraConfig defines any extra (overlay) configuration for components
+	//
+	// +kubebuilder:validation:Optional
+	ExtraConfig *MonolithicExtraConfigSpec `json:"extraConfig,omitempty"`
+}
+
+// MonolithicStorageSpec defines the storage for the Tempo deployment.
+type MonolithicStorageSpec struct {
+	// Traces defines the backend storage configuration for traces
+	//
+	// +kubebuilder:validation:Required
+	Traces MonolithicTracesStorageSpec `json:"traces"`
+}
+
+// MonolithicTracesStorageSpec defines the traces storage for the Tempo deployment.
+type MonolithicTracesStorageSpec struct {
+	// Backend defines the backend for storing traces
+	//
+	// +kubebuilder:validation:Required
+	Backend MonolithicTracesStorageBackend `json:"backend"`
+
+	// WAL defines the write-ahead logging (WAL) configuration
+	//
+	// +kubebuilder:validation:Optional
+	WAL *MonolithicTracesStorageWALSpec `json:"wal,omitempty"`
+
+	// PV defines the Persistent Volume configuration
+	//
+	// +kubebuilder:validation:Optional
+	PV *MonolithicTracesStoragePVSpec `json:"pv,omitempty"`
+}
+
+// MonolithicTracesStorageBackend defines the backend storage for traces.
+//
+// +kubebuilder:validation:Enum=memory;pv
+type MonolithicTracesStorageBackend string
+
+const (
+	// MonolithicTracesStorageBackendMemory specifies a in-memory storage backend.
+	MonolithicTracesStorageBackendMemory MonolithicTracesStorageBackend = "memory"
```

Comment:
I think we should explain more what does the in-memory mean as there is no true in-memory storage in tempo.

Reply:
I've updated the comment